pkg/controller/operands/networkAddons.go

```diff
@@ -61,12 +61,31 @@ func (h *cnaHooks) updateCr(req *common.HcoRequest, Client client.Client, exists
 		return false, false, errors.New("can't convert to CNA")
 	}
 
+	// If deployOVS annotation doesn't exists:
+	// * if on upgrade - set it to true
+	// * if not on upgrade (fresh install / manually removing the annotation) - set it to false.
+	_, annotation_exists := req.Instance.Annotations["deployOVS"]
```

Comment:
I think that if `req.Instance.Annotations` is nil then this line will crash, better rearrange it like so:
```
	// If deployOVS annotation doesn't exists:
	// * if on upgrade - set it to true
	// * if not on upgrade (fresh install / manually removing the annotation) - set it to false.
	annotationExists := false
	if req.Instance.GetAnnotations() == nil {
		req.Instance.Annotations = map[string]string{}
	} else {
		_, annotationExists = req.Instance.GetAnnotations()["deployOVS"]
	}
	if !annotationExists {
		if req.UpgradeMode {
			req.Instance.Annotations["deployOVS"] = "true"
			req.Logger.Info("deployOVS annotation did not exist in previous version and set to true in new version.")
		} else {
			req.Instance.Annotations["deployOVS"] = "false"
			req.Logger.Info("deployOVS annotation does not exist and it is being set to false.")
		}
		req.Dirty = true
	}
```

Reply:
and perhaps we should move it to a separate function `updateDeployOvsAnnotation` or something (I am bad with namings)

Reply:
if `req.Instance.Annotations == nil`, the line doesn't crash, it returns `false` to the second variable.

Reply:
are you sure? I think that if req.Instance.Annotations==nil then nil["deployOVS"] should fail.